src/core/global.c

```diff
@@ -886,6 +922,22 @@ int nn_recvmsg (int s, struct nn_msghdr *msghdr, int flags)
         *(void**) (msghdr->msg_iov [0].iov_base) = chunk;
         sz = nn_chunk_size (chunk);
     }
+    else if (msghdr->msg_iovlen > 1 && msghdr->msg_iov [0].iov_len == NN_MSG) {
+        chunk = nn_chunkref_getchunk (&msg.body);
+        
+        sz = 0;
+        for (i = 0; i != msghdr->msg_iovlen; ++i) {
+            iov = &msghdr->msg_iov [i];
```

Comment:
We can add some more checks in here to assert things such as iov->iov_len == NN_MSG before we read the length prefix, or ensuring that our accumulated iov_len's aren't > our total chunk size.